pylint/checkers/refactoring.py

```diff
@@ -261,12 +265,30 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
+        for part in node.body:
+            if isinstance(node.iter, astroid.node_classes.Name):
+                self._check_mutable_sequence_modified_in_loop(part,
+                                                              node.iter.name)
+
+    def _check_mutable_sequence_modified_in_loop(self, node, target_name):
+        if hasattr(node, 'body'):
+            for part in node.body:
+                self._check_mutable_sequence_modified_in_loop(part,
+                                                              target_name)
+        if (isinstance(node, astroid.node_classes.Expr) and
+                isinstance(node.value, astroid.node_classes.Call) and
+                isinstance(node.value.func, astroid.node_classes.Attribute) and
+                node.value.func.attrname in ['remove', 'delete', 'pop'] and
```

Comment:
It seems like you only assumes lists here. I was hoping for more generic approach that handles other mutable sequences (dictionaries, sets).

I guess it can be implemented later.